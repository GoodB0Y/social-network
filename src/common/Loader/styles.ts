import styled from 'styled-components';
import loader from './loader.svg';

const StyledLoading = styled.div<{ $size: number }>`
  background-image: url(${loader});
  background-size: contain;
  width: ${({ $size }) => $size}px;
  height: ${({ $size }) => $size}px;
  margin: 0 auto;
`;

export default StyledLoading;
