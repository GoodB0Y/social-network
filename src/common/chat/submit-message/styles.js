import styled from 'styled-components';
import { Input } from 'antd';

const { TextArea } = Input;

export const Wrap = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  box-sizing: border-box;
`;

export const TextWrap = styled.div`
  position: relative;
  display: flex;
  width: 100%;
  flex-grow: 1;
  font-family: Montserrat;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;

`;
export const Text = styled(TextArea)`
  height: 3rem;
  border: none;
  background: #FFF5E3;
  width: 100%;
  padding: 10px 40px 15px 15px;
  border-radius: 15px;
  resize: none;

  &:focus {
    box-shadow: 0 0 0 2pt red;
    border: 1px solid #fff5e3;
    outline: none;
    box-shadow: 0 0 3pt 2pt #ffb11b;
  }
`;

export const FileIcon = styled.img`
  position: absolute;
  cursor: pointer;
  top: 10px;
  padding: 5px;
  right: 0;
  transform: translateX(-100%);
`;

export const SubmitMessageIcon = styled.img`
  margin-left: 25px;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }
`;
