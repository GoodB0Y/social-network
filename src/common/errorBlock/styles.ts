import styled from 'styled-components';

const StyledError = styled.div`
  margin: 150px auto;
  width: 50%;
  border: 1px solid #FFB11B;
  border-radius:  15px;
  background-color: transparent;
  padding: 18px 36px 72px;
`;

export default StyledError;
