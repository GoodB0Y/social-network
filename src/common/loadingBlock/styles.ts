import styled from 'styled-components';
import loading from '../../img/LoadingBlock.svg';


/* eslint-disable import/prefer-default-export */
export const StyledLoading = styled.div<{$size: number}>`
    background-image: url(${loading});
    background-size: contain;
    width: ${({ $size }) => $size}px;
    height: ${({ $size }) => $size}px;
    margin: 0 auto;
`;
