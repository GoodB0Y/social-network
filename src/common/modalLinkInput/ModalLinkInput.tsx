/* eslint-disable @typescript-eslint/no-use-before-define */
import React, { useCallback, useState } from 'react';
import { Modal } from 'antd';
import {
  Button,
  Input,
  Form,
} from './styles';
import './style.scss';

interface IModalLinkInput {
    title: string;
    visible: boolean;
    setUnvisible?: () => void;
    onLinkSend: (link: string) => void;
}

const ModalLinkInput: React.FC<IModalLinkInput> = ({ visible,
  title,
  onLinkSend,
  setUnvisible }) => {
  const [value, setValue] = useState('');
  const onChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
    const innerValue = event.target.value;
    setValue(innerValue);
  }, [setValue]);
  const onSubmit = useCallback((event: React.FormEvent) => {
    event.preventDefault();
    if (setUnvisible) {
      setUnvisible();
    }
    onLinkSend(value);
  }, [value, setUnvisible, onLinkSend]);
  return (
    <Modal
      visible={visible}
      footer={null}
      title={title}
      onCancel={setUnvisible}
      centered
      className="custom-antd-modal"
    >
      <Form onSubmit={onSubmit}>
        <Input onChange={onChange} autoFocus />
        <Button>Отправить</Button>
      </Form>
    </Modal>
  );
};

export default ModalLinkInput;
