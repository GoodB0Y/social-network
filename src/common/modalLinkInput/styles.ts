import styled from 'styled-components';
import { textColor, secondaryColor, divideColorStrong } from '../../colors.module';

export const Button = styled.button`
  background-color: ${secondaryColor};
  border: 1px solid ${secondaryColor};
  padding: 5px 10px;
  color: ${textColor};
  border-radius: 5px;
`;

export const Input = styled.input<{ $error?: boolean }>`
  margin-bottom: 15px;
  border-color: ${({ $error }) => ($error ? 'red' : 'initial')};
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: white !important;
`;

export const InputTitle = styled.div`
  font-size: 15px;
  color: ${divideColorStrong};
  margin-left: 15px;
`;

export const ErrorText = styled.div`
  color: red;
  font-size: 13px;
`;
