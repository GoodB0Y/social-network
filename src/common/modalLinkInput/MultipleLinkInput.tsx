import React, { useCallback } from 'react';
import { InputTitle, Input, ErrorText } from './styles';

interface IMultipleLinkInput {
  // Передаю функцию, а не изменяю все это дело внутри
  // чтобы не зависеть от логики внутри компонента. Этот компонент - глупый, он умный компонент
  // это ModalLinkInput
  setValue: (newValue: string, index: number) => void;
  titles: string[];
  isError: boolean[];
}

const MultipleLinkInput = ({ setValue, titles, isError }: IMultipleLinkInput): JSX.Element => {
  const onChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>, index: number) => {
      const innerValue = event.target.value;
      setValue(innerValue, index);
    },
    [setValue]
  );
  const renderInputFields = () =>
    /* Отрубаю правило, потому что поля ниже никак не уникальны, кроме как по индексу в массиве */
    /* eslint-disable react/no-array-index-key */
    titles.map((title, index) => [
      <InputTitle key={`${title}_${index}`}>{title}</InputTitle>,
      isError[index] ? <ErrorText>Это поле обязательно</ErrorText> : null,
      <Input
        key={`${title}_${index}_input`}
        onChange={(event: React.ChangeEvent<HTMLInputElement>) => onChange(event, index)}
        autoFocus={index === 0}
        $error={isError[index]}
      />,
    ]);
  return <>{renderInputFields()}</>;
};

export default MultipleLinkInput;
