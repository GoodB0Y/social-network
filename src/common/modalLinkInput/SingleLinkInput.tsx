import React, { useCallback } from 'react';
import { Input, ErrorText } from './styles';

interface ISingleLinkInput {
  // Передаю функцию, а не изменяю все это дело внутри
  // чтобы не зависеть от логики внутри компонента. Этот компонент - глупый, он умный компонент
  // это ModalLinkInput
  setValue: (newValue: string) => void;
  isError: boolean;
}

const SingleLinkInput = ({ setValue, isError }: ISingleLinkInput): JSX.Element => {
  const onChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      const innerValue = event.target.value;
      setValue(innerValue);
    },
    [setValue]
  );
  return (
    <>
      <Input onChange={onChange} autoFocus $error={isError} />
      {isError && <ErrorText>Это поле обязательно</ErrorText>},
    </>
  );
};

export default SingleLinkInput;
