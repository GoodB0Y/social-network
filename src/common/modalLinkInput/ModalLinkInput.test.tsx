/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import { ModalLinkInput, IModalLinkInput } from './ModalLinkInput';
import { Form } from './styles';
import MultipleLinkInput from './MultipleLinkInput';
import SingleLinkInput from './SingleLinkInput';

describe('ModalLinkInput test', () => {
  const mockOnLinkSend = jest.fn();
  const mockSetUnvisible = jest.fn();
  const props: IModalLinkInput<string | string[]> = {
    visible: true,
    title: '',
    onLinkSend: mockOnLinkSend,
    setUnvisible: mockSetUnvisible,
  };

  test('renders propely', () => {
    const modal = shallow(<ModalLinkInput {...props} />);
    const mountModal = mount(<ModalLinkInput {...props} />);
    expect(modal).toMatchSnapshot();
    expect(mountModal).toMatchSnapshot();
  });

  describe('ModalLinkInput has single input', () => {
    const modal = mount(<ModalLinkInput {...props} />);
    const shallowModal = shallow(<ModalLinkInput {...props} />);
    const input = modal.find('input');
    test('render single input', () => {
      expect(input).toHaveLength(1);
      expect(shallowModal.find(SingleLinkInput));
    });

    describe('ModalLinkInput with single input field submits', () => {
      test('link submits with single text after input', () => {
        input.simulate('change', { target: { value: 'some text' } });
        modal.find('form').simulate('submit', { preventDefault: () => {} });
        expect(mockSetUnvisible).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledWith('some text');
      });
      /*
      test('link submits with single text after button press', () => {
        input.simulate('change', { target: { value: 'some text' } });
        modal.find('button').simulate('click', { preventDefault: () => {} });
        expect(mockSetUnvisible).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledWith('some text');
      });
      */
      test('modal dont submits because empty input', () => {
        modal.find('form').simulate('submit', { preventDefault: () => {} });
        shallowModal.find(Form).simulate('submit', { preventDefault: () => {} });
        expect(mockSetUnvisible).toHaveBeenCalledTimes(0);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(0);
        // expect(shallowModal.find(SingleLinkInput).prop('isError')).toBe(true);
      });

      beforeEach(() => {
        input.simulate('change', { target: { value: '' } });
        mockSetUnvisible.mockClear();
        mockOnLinkSend.mockClear();
      });
    });
  });

  describe('ModalLinkInput with several inputs', () => {
    const modal = mount(<ModalLinkInput {...props} title={['Hello', 'It s a me', 'Mario']} />);
    const shallowModal = shallow(<ModalLinkInput {...props} />);
    const input = modal.find('input');
    test('render 3 inputs', () => {
      expect(input).toHaveLength(3);
      expect(shallowModal.find(MultipleLinkInput));
    });
    describe('ModalLinkInput with 3 input field submits', () => {
      test('link submits with single text after input', () => {
        input.at(0).simulate('change', { target: { value: 'some text 1' } });
        input.at(1).simulate('change', { target: { value: 'some text 2' } });
        input.at(2).simulate('change', { target: { value: 'some text 3' } });
        modal.find('form').simulate('submit', { preventDefault: () => {} });
        expect(mockSetUnvisible).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledWith(['some text 1', 'some text 2', 'some text 3']);
      });
      /*
      test('link submits with single text after button press', () => {
        input.at(0).simulate('change', { target: { value: 'some text 1' } });
        input.at(1).simulate('change', { target: { value: 'some text 2' } });
        input.at(2).simulate('change', { target: { value: 'some text 3' } });
        modal.find('button').simulate('click', { preventDefault: () => {} });
        expect(mockSetUnvisible).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(1);
        expect(mockOnLinkSend).toHaveBeenCalledWith(['some text 1', 'some text 2', 'some text 3']);
      });
      */
      test('modal dont submits because empty input', () => {
        modal.find('form').simulate('submit', { preventDefault: () => {} });
        shallowModal.find(Form).simulate('submit', { preventDefault: () => {} });
        expect(shallowModal.find(MultipleLinkInput).prop('isError')).toBe(true);
        expect(mockSetUnvisible).toHaveBeenCalledTimes(0);
        expect(mockOnLinkSend).toHaveBeenCalledTimes(0);
      });

      beforeEach(() => {
        input.at(0).simulate('change', { target: { value: '' } });
        input.at(1).simulate('change', { target: { value: '' } });
        input.at(2).simulate('change', { target: { value: '' } });
        mockSetUnvisible.mockClear();
        mockOnLinkSend.mockClear();
      });
    });
  });
});
