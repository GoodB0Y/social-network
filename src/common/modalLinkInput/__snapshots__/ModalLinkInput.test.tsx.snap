// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLinkInput test renders propely 1`] = `ShallowWrapper {}`;

exports[`ModalLinkInput test renders propely 2`] = `ReactWrapper {}`;
